tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7682,8 +7713,15 @@ const Expr *SpirvEmitter::collectArrayStructIndices(
 
     {
       LowerTypeVisitor lowerTypeVisitor(astContext, spvContext, spirvOptions);
-      const uint32_t fieldIndex =
-          getFieldIndexInStruct(spirvOptions, lowerTypeVisitor, indexing);
+      const auto &astStructType =
+          /* structType */ indexing->getBase()->getType();
+      const StructType *spirvStructType =
+          lowerStructType(spirvOptions, lowerTypeVisitor, astStructType);
+      assert(spirvStructType != nullptr);
+      const uint32_t fieldIndex = getFieldIndexInStruct(
+          spirvStructType, astStructType,
+          /* fieldDecl */
+          dynamic_cast<const FieldDecl *>(indexing->getMemberDecl()));
```

Comment:
Should this be `dyn_cast`? I'm pretty sure we don't build with RTTI enabled.

Reply:
Oh thanks! Assumed dyn_cast was kind of legacy stuff, didn't knew the diff! Changing that.

Reply:
Ah, yea. LLVM's dyn_cast is a lightweight RTTI implementation. It is basically dynamic casting without typeid. I think the approach dyn_cast uses creates binaries that are around 10-15% smaller.